### ISSUE TITLE
First Product Modal pops up unexpectedly while editing homepage and is confusing

### ISSUE THREAD

**Issue by ianstewart** [Automattic/wp-calypso#65449]
This modal popped up unexpectedely about a minute into looking at my homepage after coming through the Seller Onboarding flow.

![image](https://user-images.githubusercontent.com/545779/178287859-7bbed5f9-75c4-4d1b-b234-5809b12d935c.png)

## Issues

1. Took about 30–60 seconds for it to appear.
2. The content is confusing. I didn't do anything like add a product. It just popped up.
3. If you click View Your Product, it goes to the homepage. Also confusing.

**Comment by blackjackkent** [comment]
My guess off the top of my head is that this is accidentally triggering on the first save of any sort - manual or auto - rather than just manual.

Tasks:
- Trigger only on manual save
- Trigger only if user has actually updated payments block from initial dummy data(?)

**Comment by mreishus** [comment]
Autosave only:
* #65707

Two optimizations I noticed while I was looking around:
* #65708
* #65712

I'll take a look at narrowing the condition to being only when a payments block is updated, but I'm not too sure if it's easily done.

**Comment by mreishus** [comment]
Checking for a payments block to be updated and properly set up ended up being a bit of a rabbit hole that I wasn't able to get working yet. (I tried approaches using `getEntityRecordEdits` and `getReferenceByDistinctEdits`).

However, I was able to find a decent compromise in my opinion: Require a payments block to be selected at least once before displaying the modal. This stops the user from seeing it in the "load homepage, edit title, press save" scenario, and ensures they will see it when they are legitimately setting up a block. It's not perfect - they still might trigger it if they click around everywhere before saving - but it seems better than before.

* #65761
